### PR TITLE
add support for claiming bech32 bitcoin addresses

### PR DESCRIPTION
**Pull request by xgess**
* add bech32 (segwit native) support for bitcoin addresses
* fix a bug with casing bech32 addresses
* put the ability to sign new links behind a feature flag

this depends on https://github.com/keybase/node-bitcoyne/pull/3 and https://github.com/keybase/keybase/pull/3657

### CODE REVIEW

File: go/engine/cryptocurrency.go [hunk 1]
```diff
@@ -48,13 +49,30 @@ func (e *CryptocurrencyEngine) SubConsumers() []libkb.UIConsumer {
...
+func normalizeAddress(address string) string {
+	switch strings.ToLower(address)[0:3] {
+	case "bc1", "zs1":
+		// bech32 addresses require that cases not be mixed
+		// (an error which will be caught downstream in validation),
+		// but the spec is otherwise case-insensitive. so if it's
+		// passed in as uppercase, then we downcase it right away to
+		// ensure that everything we do with the address (e.g. sign it into
+		// a sigchain link) will be consistent and checksum correctly.
+		if strings.ToUpper(address) == address {
```
Comment (zapu) [inline go/engine/cryptocurrency.go]: so what happens if it's mixed case? do we fail somewhere?

**Comment by maxtaco** [comment]
I'm not following the full narrative here, but it's important to feature-flag posting off for a few releases until enough readers of segwit are out in the wild

File: go/libkb/cryptocurrency.go [hunk 2]
```diff
@@ -156,7 +176,7 @@ func BtcAddrCheck(s string, _ *BtcOpts) (version int, pkhash []byte, err error)
 	if err != nil {
 		return version, pkhash, err
 	}
-	if typ != CryptocurrencyTypeBTC && typ != CryptocurrencyTypeBTCMultiSig {
+	if typ != CryptocurrencyTypeBTC && typ != CryptocurrencyTypeBTCMultiSig && typ != CryptocurrencyTypeBTCSegwit {
 		return int(CryptocurrencyTypeNone), nil, errors.New("only support BTC vanila and multisig")
```
Comment (zapu) [inline go/libkb/cryptocurrency.go]: should this message be changed?

File: go/engine/cryptocurrency.go (hunk 1 above)
Comment (xgess) [reply go/engine/cryptocurrency.go]: yeah. it's recognized as invalid on the server and an error will bubble up.

File: go/libkb/cryptocurrency.go (hunk 2 above)
Comment (xgess) [reply go/libkb/cryptocurrency.go]: yes. done.

**Pull request merged**